packages/component-base/src/polylit-mixin.js

```diff
@@ -203,8 +224,10 @@ const PolylitMixinImplementation = (superclass) => {
         this.$ = {};
       }
 
-      this.renderRoot.querySelectorAll('[id]').forEach((node) => {
-        this.$[node.id] = node;
+      [...Object.values(this.$), this.renderRoot].forEach((node) => {
```

Comment:
Do I understand it correctly that this change is intended to cover cases when the element already placed in `this.$` is an overlay that is being teleported? It would be good to add a comment and a test for this.

Reply:
Yes, that's correct, and I decided to extract this change into a separate PR: https://github.com/vaadin/web-components/pull/8299